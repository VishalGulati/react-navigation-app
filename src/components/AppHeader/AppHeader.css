header {
  text-align: left;
  padding: 10px;
  background-color: #4a569d;
  width: 100%;
  position: fixed;
  top: 0;
  z-index: 11;
}
