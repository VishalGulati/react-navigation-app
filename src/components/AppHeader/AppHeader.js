import React from 'react';
import './AppHeader.css';

/**
 * The application header
 */
const AppHeader = () => {
  return <header>React Navigation App</header>;
};

export default AppHeader;
