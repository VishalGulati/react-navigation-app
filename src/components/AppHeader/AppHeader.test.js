import React from 'react';
import ReactDOM from 'react-dom';
import AppHeader from './AppHeader';
import { shallow } from 'enzyme';
const wrapper = shallow(<AppHeader />);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AppHeader />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Component has header element', () => {
  expect(wrapper.find('header').length).toEqual(1);
});

it('snapshot testing', () => {
  expect(wrapper).toMatchSnapshot();
});
