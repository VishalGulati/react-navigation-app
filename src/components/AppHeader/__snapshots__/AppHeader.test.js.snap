// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppHeader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "React Navigation App",
    },
    "ref": null,
    "rendered": "React Navigation App",
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "React Navigation App",
      },
      "ref": null,
      "rendered": "React Navigation App",
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
